recipes-bsp/u-boot/files/0010-dts-iot2050-Support-new-IOT2050-SM-variant.patch

```diff
@@ -0,0 +1,447 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Baocheng Su <baocheng.su@siemens.com>
+Date: Wed, 29 Nov 2023 23:35:51 +0800
+Subject: [PATCH] dts: iot2050: Support new IOT2050-SM variant
+
+the dts file for IOT2050-SM variant is copied from kernel side without
+any change.
+
+Signed-off-by: Baocheng Su <baocheng.su@siemens.com>
+---
+ arch/arm/dts/Makefile                         |   4 +-
+ arch/arm/dts/k3-am65-iot2050-boot-image.dtsi  |   2 +-
+ .../k3-am6548-iot2050-advanced-sm-u-boot.dtsi |   1 +
+ .../arm/dts/k3-am6548-iot2050-advanced-sm.dts | 346 ++++++++++++++++++
+ board/siemens/iot2050/board.c                 |  15 +-
+ 5 files changed, 365 insertions(+), 3 deletions(-)
+ create mode 120000 arch/arm/dts/k3-am6548-iot2050-advanced-sm-u-boot.dtsi
+ create mode 100644 arch/arm/dts/k3-am6548-iot2050-advanced-sm.dts
+
+diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
+index 85fd5b1157b19a14a2ee908f1c62ce1e99c241c1..0edd5b73a414aaa000c93efe163d21a9ff91dc6c 100644
+--- a/arch/arm/dts/Makefile
++++ b/arch/arm/dts/Makefile
+@@ -1316,7 +1316,9 @@ dtb-$(CONFIG_SOC_K3_AM654) += \
+ 	k3-am6548-iot2050-advanced-pg2.dtb \
+ 	k3-am6548-iot2050-advanced-m2.dtb \
+ 	k3-am6548-iot2050-advanced-m2-bkey-usb3-overlay.dtbo \
+-	k3-am6548-iot2050-advanced-m2-bkey-ekey-pcie-overlay.dtbo
++	k3-am6548-iot2050-advanced-m2-bkey-ekey-pcie-overlay.dtbo \
++	k3-am6548-iot2050-advanced-sm.dtb
++
+ dtb-$(CONFIG_SOC_K3_J721E) += k3-j721e-common-proc-board.dtb \
+ 			      k3-j721e-r5-common-proc-board.dtb \
+ 			      k3-j7200-common-proc-board.dtb \
+diff --git a/arch/arm/dts/k3-am65-iot2050-boot-image.dtsi b/arch/arm/dts/k3-am65-iot2050-boot-image.dtsi
+index 64318d09cf0a6f6b26daa5d74dfe04b163e0a4cc..3b991baac8eeace9e4f74a25db71ae013940a1a4 100644
+--- a/arch/arm/dts/k3-am65-iot2050-boot-image.dtsi
++++ b/arch/arm/dts/k3-am65-iot2050-boot-image.dtsi
+@@ -229,7 +229,7 @@
+ 		};
+ 
+ 		fit@380000 {
+-			fit,fdt-list-val = "k3-am6528-iot2050-basic-pg2", "k3-am6548-iot2050-advanced-pg2", "k3-am6548-iot2050-advanced-m2";
++			fit,fdt-list-val = "k3-am6528-iot2050-basic-pg2", "k3-am6548-iot2050-advanced-pg2", "k3-am6548-iot2050-advanced-m2", "k3-am6548-iot2050-advanced-sm";
```

Comment:
Does the dtc allow us to wrap this rather long line?

Reply:
Fixed in #509